src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php

```diff
@@ -42,12 +54,186 @@ public function createNew()
     }
 
     /**
-     * Get order repository.
+     * @param Request $request
      *
+     * @return Response
+     */
+    public function addAction(Request $request)
```

Comment:
Is there any difference between cart item and an order item? If no, I would suggest to merge them together. So we should remove CartItemType and dispatch an `order_item` events here. We can stay with `$cart` and `$order` separation, but here I would vote to keep them together.